src/System.Management.Automation/security/SecuritySupport.cs

```diff
@@ -1312,99 +1241,52 @@ private void ResolveFromPath(SessionState sessionState, ResolutionPurpose purpos
+                    storeCU.Open(OpenFlags.ReadOnly);
+                    X509Certificate2Collection storeCerts = storeCU.Certificates;
+
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                     {
-                        certificates.Add(certificateObject);
+                        using (var storeLM = new X509Store("my", StoreLocation.LocalMachine))
+                        {
+                            storeLM.Open(OpenFlags.ReadOnly);
+                            storeCerts.AddRange(storeLM.Certificates);
+                        }
                     }
-                }
-            }
-            catch (SessionStateException)
-            {
-                // If we got an ItemNotFound / etc., then this didn't represent a valid path.
-            }
-
-            List<X509Certificate2> certificatesToProcess = new List<X509Certificate2>();
-            foreach (PSObject certificateObject in certificates)
-            {
-                X509Certificate2 certificate = certificateObject.BaseObject as X509Certificate2;
-                if (certificate != null)
-                {
-                    certificatesToProcess.Add(certificate);
-                }
-            }
-
-            ProcessResolvedCertificates(purpose, certificatesToProcess, out error);
-        }
-
-        private void ResolveFromSubjectName(SessionState sessionState, ResolutionPurpose purpose, out ErrorRecord error)
-        {
-            Collection<PSObject> certificates = new Collection<PSObject>();
-            WildcardPattern subjectNamePattern = WildcardPattern.Get(_identifier, WildcardOptions.IgnoreCase);
 
-            try
-            {
-                // Get first from 'My' store, then 'LocalMachine'
-                string[] certificatePaths = new string[] {
-                        "Microsoft.PowerShell.Security\\Certificate::CurrentUser\\My",
-                        "Microsoft.PowerShell.Security\\Certificate::LocalMachine\\My" };
+                    certificatesToProcess.AddRange(storeCerts.Find(X509FindType.FindByThumbprint, _identifier, validOnly: false));
 
-                foreach (string certificatePath in certificatePaths)
-                {
-                    foreach (PSObject certificateObject in sessionState.InvokeProvider.ChildItem.Get(certificatePath, false))
+                    if (certificatesToProcess.Count == 0)
                     {
-                        if (subjectNamePattern.IsMatch(certificateObject.Properties["Subject"].Value.ToString()))
+                        foreach (var cert in storeCerts)
                         {
-                            certificates.Add(certificateObject);
+                            if (subjectNamePattern.IsMatch(cert.Subject))
+                            {
+                                certificatesToProcess.Add(cert);
+                            }
```

Comment:
@mikeTWC1984 If you ask about this IsMatch() it is ok to preserve the old behavior until we get a negative feedback.

Reply:
This one is OK. It's bit counterintuitive to use wildcard (and directory) when expecting single item, but helps when typing those names  manually.  After going back and forth with this, I realized omitting  CN= prefix  in subject name would be very useful, so I added one extra IsMatch for simple name too